cpp/open3d/t/io/file_format/FileTXT.cpp

```diff
@@ -0,0 +1,273 @@
+// The MIT License (MIT)
+//
+// Copyright (c) 2018-2021 www.open3d.org
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+// FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+// IN THE SOFTWARE.
+// ----------------------------------------------------------------------------
+
+#include <cstdio>
+
+#include "open3d/core/Dtype.h"
+#include "open3d/core/Tensor.h"
+#include "open3d/io/FileFormatIO.h"
+#include "open3d/t/io/PointCloudIO.h"
+#include "open3d/utility/FileSystem.h"
+#include "open3d/utility/Logging.h"
+#include "open3d/utility/ProgressReporters.h"
+
+namespace open3d {
+namespace t {
+namespace io {
+
+open3d::io::FileGeometry ReadFileGeometryTypeTXT(const std::string &path) {
+    return open3d::io::CONTAINS_POINTS;
+}
+
+bool ReadPointCloudFromTXT(const std::string &filename,
+                           geometry::PointCloud &pointcloud,
+                           const open3d::io::ReadPointCloudOption &params) {
+    try {
+        pointcloud.Clear();
+
+        utility::filesystem::CFile file;
+        if (!file.Open(filename, "r")) {
+            utility::LogWarning("Read TXT failed: unable to open file: {}",
+                                filename);
+            return false;
+        }
+        utility::CountingProgressReporter reporter(params.update_progress);
+        reporter.SetTotal(file.GetFileSize());
+
+        std::string format_txt =
+                utility::filesystem::GetFileExtensionInLowerCase(filename);
+
+        int64_t num_points = file.GetNumLines();
+        int64_t num_attributes;
+
+        if (format_txt == "xyz") {
+            num_attributes = 3;
+        } else if (format_txt == "xyzi") {
+            num_attributes = 4;
+        } else if ((format_txt == "xyzn") || (format_txt == "xyzrgb")) {
+            num_attributes = 6;
+        } else {
+            utility::LogWarning("The format of TXT is not supported");
+            return false;
+        }
+
+        core::Tensor pcd_buffer({num_points, num_attributes}, core::Float64);
```

Comment:
After thinking about it again, it is best to keep everything `Float32` by default. This is because all creation APIs of the tensor TriangleMesh and PointCould uses Float32. Could you change everything here to float32?

Reply:
Now the other files io in tensor module use `double` as default, so the tests of io also use `double` as default. If I use `float` as default in FileTXT, it cannot pass the tests. 
I think we may keep `double` as default this time and modify the default dtype from double to float in the relevant files in next pr. What do you think about it? @yxlao